### ISSUE TITLE
[Feature Request] Disable "shadowing" of methods

### ISSUE THREAD

**Issue by HLWeil** [CSBiology/DynamicObj#24]
Given the following class

```fsharp
type Person(name : string) =
    
    inherit DynamicObj()

    let mutable name = name

    member this.Name
        with get() = name
        and set(value) = name <- value

    member this.Walk () = printfn "%s is walking" name
```

You can currently set a dynamic property with the same name as an existing method.
```fsharp
let p = Person("John")

p.SetValue("Walk", "Shadowing")

p.Walk() // John is walking

p.GetValue("Walk") // val it: obj = "Shadowing"
```

Should this work, or fail? @kMutagene

**Comment by kMutagene** [comment]
Can we even transpile a lookup of the methods or is than non-transpilable reflection?

**Comment by HLWeil** [comment]
Yes this is transpilable as far as I know. In principle it could the same logic as the `Systen.ICloneable.Clone()` lookup you implemented for the DeepCopy method. E.g. 

https://github.com/CSBiology/DynamicObj/blob/073da2dcf657200a7bf3f65028baa93afa0d7abc/src/DynamicObj/FableJS.fs#L163

**Comment by kMutagene** [comment]
On another Note, this currently removes the method in transpiled Code right? I think this is expected behaviour in js/py right? So when we prevent this it gets less idiomatic in those languages
